front/src/components/QrReader/components/QrReader.tsx

```diff
@@ -33,6 +35,14 @@ function QrReader({
   onScan,
   onFindBoxByLabel,
 }: IQrReaderProps) {
+  const { user } = useAuth0();
+
+  // Check if user has manage_inventory permission
+  const hasManageInventoryPermission = useMemo(() => {
+    if (!user || !user[JWT_ABP]) return false;
+    return user[JWT_ABP].includes("manage_inventory");
+  }, [user]);
```

Comment:
this block also appears in QrReaderMultiBoxContainer.tsx and BoxView.tsx. Please refactor it into a dedicated hook in hooks.ts that takes a permission string as input.